src/ParameterTypes/passive_ren.jl

```diff
@@ -92,14 +99,29 @@ function direct_to_explicit(ps::PassiveRENParams{T}, return_h=false) where T
     # Currently converts to Hermitian to avoid numerical conditioning issues
     M = _M_pass(X3, Y3, ϵ)
 
-    D22 = ν*I + M
-    D21_imp = D21 - D12_imp'
+    if ρ == 0
+        # For ρ==0 case, I(SI)P model
+        D22 = ν*I + M
+        D21_imp = D21 - D12_imp'
+
+        𝑅  = _R_pass(D22, ν, ρ) 
+        Γ2 = _Γ2_pass(C2, D21_imp, B2_imp, 𝑅)
 
-    𝑅  = _R_pass(D22, ν)
-    Γ2 = _Γ2_pass(C2, D21_imp, B2_imp, 𝑅)
+        H = x_to_h(X, ϵ, polar_param, ρ_polar) + Γ2
+    else    
+        # For ρ!=0 case, ISOP model
+        D22 = 1/ρ *(I+M)\I
+        C2_imp = (D22'*(-2ρ*I) + I)*C2
+        D21_imp = (D22'*(-2ρ*I) + I)*D21 - D12_imp'
```

Comment:
@MrstupidJ I think it would be better to create a small function for each of these computations:
```julia
D22 = 1/ρ *(I+M)\I
C2_imp = (D22'*(-2ρ*I) + I)*C2
D21_imp = (D22'*(-2ρ*I) + I)*D21 - D12_imp'
```
Putting more complex operations inside a function helps to speed up the backwards pass during back-propagation.

Reply:
Maybe leave `D22` and I will make small function for the other two. 

`D22` has quite different parameterisation. For `C_2imp` and `D2_imp`, they are still different, but they have quite simple expression in I(SI)P case. Those complex operations you pointed out were ISOP case.

Reply:
I think it's still worth it even for `D22` since it still involves computing a matrix inverse.